etcdctlv3/command/global.go

```diff
@@ -115,6 +117,15 @@ func newClientCfg(endpoints []string, dialTimeout time.Duration, scfg *secureCfg
 		cfg.TLS = &tls.Config{}
 	}
 
+	// If the user wants to skip TLS verification then we should setup
+	// the tls configuration with InsecureSkipVerify flag set.
+	if scfg.insecureSkipVerify {
+		if cfg.TLS == nil {
```

Comment:
if tls == nil, it means etcdctl is using insecure connection. in this case, we should probably just ignore skipverify thing. if we set it as line 124, we basically force user to use secure connection when the flag is provided.